src/sds.c

```diff
@@ -204,8 +207,24 @@ sds sdsdup(const_sds s) {
 /* Free an sds string. No operation is performed if 's' is NULL. */
 void sdsfree(sds s) {
     if (s == NULL) return;
+    if (sdsType(s) == SDS_TYPE_32_SHARED) {
+        SDS_HDR_VAR(32shared, s);
+        sdsReleaseShared(sh);
+        return;
+    }
     s_free_with_size(sdsAllocPtr(s), sdsAllocSize(s));
 }
+/* return 1 if the shared sds is freed otherwise 0 */
+int sdsReleaseShared(sdshdrshared *sh) {
```

Comment:
Why do we need to expose this function in the API? The caller can use `sdsfree` instaed and it does the ref-counter logic for the shared ones. We can slim the API and skip this function.